core/sql/src/main/java/org/trafodion/sql/udr/TupleInfo.java

```diff
@@ -1079,19 +1230,48 @@ else if (t.getByteLength() == 8)
             row_.putLong(t.getDataOffset(), val);
             break;
 
+        case TINYINT_UNSIGNED:
+            if (val < 0 || val > Byte.MAX_VALUE)
+                if (val > (2 * Byte.MAX_VALUE + 1))
+                    throw new UDRException(
+                        38900,
+                        "Overflow/underflow when assigning to TINYINT UNSIGNED type");
+                else
```

Comment:
Will this handle underflow correctly? If val < 0 is an exception supposed to be thrown?